src/components/Header/Header.js

```diff
@@ -40,6 +49,6 @@ class Header extends Component {
       </div>
     );
   }
-}
+});
```

Comment:
This is fairly canonical, but I have some comments:

1. The navigation elements can and should be separated out into a component independent of Header. Call it, say, `Navigation`, and invoke it here. It will encapsulate all the nav stuff, including the state and handlers that make the active selection behave. This is very much in the spirit of React programming. Never hesitate to create a component that can encapsulate and isolate complexity, particularly if that includes state. State should always be as far down the hierarchy as possible.
2. Make `updateNav()` and `handleSelect()` _methods of the class_ (your new `Navigation` class). It looks odd to have them inside `render()`; they look as if they are (although they are not, actually) modifying state inside `render`, which is verboten.

Reply:
Another thought. I think this won't correct set the navigation when you drive directly to a route via URL. Is there a way you can use the `activeHref` property of `Nav` and always set it from the current path? I think that ought to be straightforward but I don't know it for sure. It sounds more robust, and it saves you having to manage the state explicitly in the `Navigation` component -- the router will do that for you.

Reply:
Great suggestions, though I think that I'm out of time on this issue. Going to write up an enhancement issue, point to this discussion and defer to the work to our next round of revisions.